src/main/java/htsjdk/variant/variantcontext/VariantContextBuilder.java

```diff
@@ -339,26 +366,55 @@ private void makeAttributesModifiable() {
      * filters can be <code>null</code> -&gt; meaning there are no filters
      *
      * @param filters Set of strings to set as the filters for this builder
+     *                This set will be copied so that external set can be
+     *                safely changed.
      * @return this builder
      */
     public VariantContextBuilder filters(final Set<String> filters) {
-        this.filters = filters;
+        if (filters == null) {
+            unfiltered();
+        } else {
+            makeFiltersModifiable();
```

Comment:
I think these lines
```
            makeFiltersModifiable();
            this.filters.clear();
            this.filters.addAll(filters);
            toValidate.add(VariantContext.Validation.FILTERS);
```
can be replaced with
```
this.filtersCanBeModified = true;
filtersAsIs(new HashSet<>(filters));
```

There's a lot of overlapping methods so it's hard for me to keep clear exactly what method does what thing. On the other hand, maybe it's OK to merge it up without cleaning it up too much...

Reply:
:+1: